collect_app/src/main/java/org/odk/collect/android/activities/FormDownloadListActivity.java

```diff
@@ -597,33 +572,55 @@ public void formListDownloadingComplete(HashMap<String, ServerFormDetails> resul
             toggleButtonLabel(toggleButton, listView);
 
             if (viewModel.isDownloadOnlyMode()) {
-                //1. First check if all form IDS could be found on the server - Register forms that could not be found
+                performDownloadModeDownload();
+            }
+        } else {
+            switch (exception.getType()) {
+                case FETCH_ERROR:
+                case UNKNOWN_HOST:
+                    String dialogMessage = new FormApiExceptionMapper(getResources()).getMessage(exception);
+                    String dialogTitle = getString(R.string.load_remote_form_error);
+
+                    if (viewModel.isDownloadOnlyMode()) {
+                        setReturnResult(false, getString(R.string.load_remote_form_error), viewModel.getFormResults());
```

Comment:
Why not `dialogMessage`? That seems a lot more useful.

Reply:
Good point. Will change.